### PR TITLE
Allow capitals in email regex

### PR DESCRIPTION
**Pull request by robotbill**
Updates the regular expression to allow capital letters so that we can
use it for frontend validation without rejecting valid emails.

### CODE REVIEW

**Review by mockdeep — APPROVED** [review]

File: lib/email_repair/constants.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ class << self
       def email_regex
         local_part_regex = "[#{valid_chars}]" \
           "([#{valid_chars_with_dot}]*[#{valid_chars}])?"
-        /#{local_part_regex}@(?:[a-z0-9\-]+\.)+(?:[a-z]{2,24})/
+        /#{local_part_regex}@(?:[a-zA-Z0-9\-]+\.)+(?:[a-zA-Z]{2,24})/
```
Comment (AlexWayfer) [inline lib/email_repair/constants.rb:7]: Don't you think the `/i` is better?

**Review by AlexWayfer — CHANGES_REQUESTED** [review]
Dude, the README literally says first of all:

> downcases the email

https://github.com/InformedK12/email-repair/blob/master/README.md#commonmistakerepair

image

**Comment by mockdeep** [comment]
@AlexWayfer that's what we do when the email is passed into `EmailRepair`. For validation *before* we run it through `EmailRepair`, though, we want to be a little more permissive. It's valid to have different case in emails, so we want to allow users to input it. We're using it to validate using a pattern on the email field:

```haml
= form.email_field(:email, pattern: EmailRepair::Constants.email_regex.source)
```

File: lib/email_repair/constants.rb (hunk 1 above)
Comment (mockdeep) [reply lib/email_repair/constants.rb:7]: Worth a try!

**Comment by AlexWayfer** [comment]
> @AlexWayfer that's what we do when the email is passed into `EmailRepair`.

I don't understand. If you call `downcase` before `EmailRepair` — it's not a reason to remove the same from the common (public) lib.



> For validation _before_ we run it through `EmailRepair`, though, we want to be a little more permissive. It's valid to have different case in emails, so we want to allow users to input it.

1. At least, it worths to change the README at least.
2. If the first goal of a lib is to downcase ("normalize", not validate) email — probably you should not change it.
3. In the end, it can be an option.

**Comment by mockdeep** [comment]
@AlexWayfer `EmailRepair` still [downcases the email internally](https://github.com/InformedK12/email-repair/blob/817bbbb56481ce56be9b6bfcd1272a6fd18d8700/lib/email_repair/mechanic.rb#L44). The only thing changing is the regex that determines whether an email is valid. I don't think there's really a need for this to be an option. The only thing affected by this is if you use `EmailRepair::Constants.email_regex` directly.

**Pull request merged**